### DISCUSSION TITLE
OSPOlogy Follow-up: Building an OSPO in the Energy Sector

### DISCUSSION THREAD

**Discussion by anajsana** [todogroup/ospology#63] [OSPO Stories]
Discuss with the community topics related to the upcoming OSPOlogy meeting:

📅 March, 2nd [Building an OSPO in the Energy Sector](https://community.linuxfoundation.org/events/details/lfhq-todo-group-presents-building-an-ospo-in-the-energy-sector-the-alliander-experiences/)

In this open space, people can:

* Ask questions prior to the meeting starts
* Share notes from the conversation
* Make proposals for new initiatives that can help the OSPO Ecosystem
* Add useful links and resources related to the topic
* Connect with mentors involved in OSPOs and/or Open Source Management

**Comment by nicorikken** [comment]
Slides for the Alliander experience part of the presentation today:
[OSPOlogy_Building an OSPO_The Alliander Experiences.pdf](https://github.com/todogroup/ospology/files/8170939/OSPOlogy_Building.an.OSPO_The.Alliander.Experiences.pdf)
 
If you would like to know more:

- [Alliander Open Source website](https://www.alliander.com/en/open-source/)
- [Alliander GitHub Open Source organization](https://github.com/alliander-opensource/)
- [LF Energy website](https://www.lfenergy.org/)

**Reply by anajsana** [reply]
FYI: The recording session of the first presentation minutes will be shared soon!

**Comment by LayZeeDK** [comment]
Are there any technology preferences/recommendations for LF Energy software?

**Reply by nicorikken** [reply]
This very much depends on the project. On embedded devices a lower level programming language like C/C++/Rust and on larger applications a higher level language like Python or Java. Most important is the interoperability between projects, but comes from protocols and standards.

**Reply by LayZeeDK** [reply]
Are these Alliander's or LF Energy's preferences/suggestions? 🙂 Which programming languages are used the most in LF Energy products?

**Reply by nicorikken** [reply]
I think this is even more generic, a programming language has to fit the purpose. There are no preferences by LF Energy, also because most of these project build on existing projects or have a longer history as an internal project by one of the members. As such it doesn't make sense to dictate programming languages.

Most-used I guess is Java. But again, this depends a lot on the context. [OperatorFabric](https://www.lfenergy.org/projects/operatorfabric/) and [GXF](https://www.lfenergy.org/projects/gxf/) are larger applications responsible for concentrating data and are built in Java. [SOGNO](https://www.lfenergy.org/projects/sogno/) and [OpenSTEF](https://github.com/OpenSTEF/openstef) are more focused on analytics processing and are written in Python. [Fledgepower](https://www.lfenergy.org/projects/fledgepower/) runs on edge devices and is written in C++

As said the main recommendation by LF Energy is to make sure it works together as best as possible. And if libraries can be used in multiple projects, that too is an advantage.

**Comment by LayZeeDK** [comment]
Which standard-defined data formats such as CIM are applicable across energy industries/types? I'm thinking heat energy, potential energy, electricity, Power-to-X.

**Reply by anajsana** [reply]
well-known standards used: CIM, IEC 61850, OpenADR

**Reply by LayZeeDK** [reply]
Thank you, Ana

**Reply by nicorikken** [reply]
Also, these standards do have gaps. A CIM-like model for the domain of natural gas is lacking for example.

**Reply by LayZeeDK** [reply]
@nicorikken Yes, we're looking to build an open source product for energy traceability regardless of energy form.

**Comment by LayZeeDK** [comment]
Where is the maturity model mentioned in the presentation?

**Reply by lijiangsheng1** [reply]
https://linuxfoundation.org/tools/the-evolution-of-the-open-source-program-office-ospo/
here

**Reply by LayZeeDK** [reply]
Thank you

**Comment by LayZeeDK** [comment]
Why is LF Energy software spread across so many GitHub organizations? Have we considered the impact of projects living in company-tied GitHub organizations, regarding ownership?

**Reply by LayZeeDK** [reply]
Answer: A GitHub organization per LF Energy project, maintained by LF Energy. A neutral ground.

**Comment by anajsana** [comment]
Jari asks:

Was it easier to sell co-creation and participating OSS projects to Alliander top management because typically there is a natural local monopoly in the energy business?

**Reply by nicorikken** [reply]
In the Netherlands grid operators are publicly owned and operate as a public body. This is different from the energy companies selling energy. So yeah grid operators have sort of a monopoly in the sense that they are have security of their customers. At the same time we are regulated to offer quality and reliability at a low cost. So co-creation certainly makes sense, to offer more for less. But similarly I think that Open Source would also benefit companies with a stronger commercial interest. Considering the huge challenges in the sector of digitization and embedding renewable energy, we simply need to collaborate. And Open Source offers a great way of collaborating.

**Comment by anajsana** [comment]
Jari asks:

What tools are you using for license checking process, ORT? Any commercial SCA tools in use?

**Reply by nicorikken** [reply]
At the moment we use https://github.com/oss-review-toolkit/ on a per-repository level and GitHub Insights to get an overview across all our projects. GitHub Insights comes with GitHub and gives a rough but not entirely flawless overview that is helpful to get an overview and pinpoint potential issues. ORT helps us get a more detailed analysis at a greater level of accuracy. The necessary tooling for ORT is still a challenge, so adoption is still limited.

**Reply by juliancoccia** [reply]
Here is another tool you might want to throw in the mix. SCANOSS performs identifications of undeclared software by making a comparison against a public Open Source Knowledgebase. Here is a quick recipe:

pip3 install scanoss
scanoss-py scan directory/

**Reply by nicorikken** [reply]
Indeed, I like the publication of a public repository to check against. I always explain it as a parallel with schools where work can be checked for plagiarism against previously submitted work. I expect the analysis for copied source code to result in a lot of false-positives, for which the legal risk is very low. So in general I consider this analysis of licenses a growth path. I think we should first focus on getting the overview and getting the organization accustomed to checking these tings, before dialing up the intensity of the checks.

**Comment by anajsana** [comment]
Jari asks:

How big is Alliander OSPO and do you have a virtual team too how big is the virtual team?

**Reply by Jonasvdbo** [reply]
The OSPO core team consists of 5 people that meet on a bi-weekly basis and the day-to-day operations of the OSPO are run by Nico and myself. Furthermore, we have an OSPO expert team who are involved when necessary. This included other experts from the IT department and experts from the legal and communications departments.

**Comment by LayZeeDK** [comment]
Who should I reach out to in order to discuss the process of donating products to LF Energy?

**Reply by amye** [reply]
https://wiki.lfenergy.org/display/HOME/New+Project+Proposals+Process has details for this, @jmertic may also be helpful here.

**Reply by LayZeeDK** [reply]
Thank you, Amye

**Comment by LayZeeDK** [comment]
Can you describe your instructing/coaching/educating efforts? That's very interesting.

**Reply by nicorikken** [reply]
We have held the generic open source webinar to have a general introduction to colleagues. We offer an ever growing wiki with lots of information and links. And we have periodic meetings with all open source projects to check how it is going and come up with tasks to improve. These can be improvements on many fronts, like project organization, promotion, documentation, licenses, etc. Lastly we try to be as open to questions as possible, so we have helped colleagues on multiple occasions navigate the world of open source.

**Comment by anajsana** [comment]
@LayZeeDK asks: Do you have a public open source portal similar to your inner source portal?

**Reply by LayZeeDK** [reply]
Thanks for copying my question to this discussion 🙂

**Reply by Jonasvdbo** [reply]
No, we don't have a public open-source portal similar to your inner source portal. Our open-source projects can be found at github.com/alliander-opensource and at lfenergy.org

**Comment by Jonasvdbo** [comment]
Nico and I answered all remaining questions. Thank you for attending and feel free to reach out if you have more questions!

**Reply by LayZeeDK** [reply]
Thank you both
